include/serenity.proto

```diff
@@ -0,0 +1,129 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+* Proto for serenity framework. Describes Input and the Output of the whole flow
+* and internal messages
+**/
+
+//Because serenity is a separate project, you have to copy mesos.proto to same directory
+import "mesos.proto";
+
+package mesos;
+
+option java_package = "org.apache.mesos";
+option java_outer_classname = "Protos";
+
+/**
+* NOTE: These proto buffers should be upstreamed - it defines
+* external communication (slave -> serenity -> slave)
+**/
+
+/**
+* Contains resource usage for running executors
+**/
+message Usage {
+ repeated ResourceUsage resource_usage = 1;
+}
+
+
+/**
+* Contains slack resources which can be offered for BE tasks
+**/
+message Slack {
+  repeated Resource resources = 1;
+}
+
+
+/**
+* Describes the type of correction.
+**/
+enum CorrectionAction {
+  KILL = 1;
+  FREEZE = 2;
+  RESIZE = 3;
+  MIGRATE = 4;
+}
+
+
+/**
+* Necessary union to define executor or task
+**/
+message JobID {
+  enum Type {
+    EXECUTOR = 1;
+    TASK = 2;
+  }
+  // Identifies which field is filled in
+  required Type type = 1;
+  //One of the following will be filled in
+  optional ExecutorID executor = 2;
+  optional TaskID task = 3;
+}
+
+
+/**
+* Describes a message between QoS policy executor and slave.
+* QoS executor makes a decision based on Contention messages
+* and some policies. In the end it sends information to slave
+* informing that some correction has to be made.
+**/
+message Correction {
+  repeated JobID jobId = 1;
+  optional CorrectionAction action = 2;
+  optional Text reason = 3;
+  optional double timestamp = 4;
+  optional Labels labels = 5;
+}
+
+
+/**
+* Describes a reaction of slave on QoS correction message.
+* It contains information about what action has been done
+* and what executors or tasks were affected.
+**/
+message Reaction {
+  optional CorrectionAction action = 1;
+  repeated JobID jobId = 2;
+  optional double timestamp = 3;
+}
+
+
+/**
+* NOTE: These proto buffers are for internal communication
+**/
+
+/**
+* Describes resource contention and noisy neighbour situation.
+* It describes certain type of resource starvation (cpu, cache, io)
+* and the victim of starvation (only a HP task is applicable).
+* If possible, it also points the probably aggressor executor or task.
+**/
+message Contention{
+  enum Type {
+    CACHE = 1;
+    CPU = 2;
+    IO = 3;
+    NETWORK = 4;
+  }
+  optional JobID victim = 1;
+  optional JobID aggressor = 2;
+  optional Type type = 3;
+  optional double timestamp = 4;
+  optional double severity = 5;
```

Comment:
I would rather advocate for varint (like from 1 to 100) here. I don't think we will need a accurate precision up to (let's say) 5th decimal place.